diff --git a/lowlevel/Makefile.Linux b/lowlevel/Makefile.Linux
index e8064c0..f42e158 100644
--- a/lowlevel/Makefile.Linux
+++ b/lowlevel/Makefile.Linux
@@ -51,7 +51,7 @@ sources_uninstall:
 install: all libimas.so_install libimas.a_install pkgconfig_install sources_install | $(libdir) $(includedir) $(docdir)/dev/lowlevel $(bindir)
 	$(INSTALL_DATA) matlab_adapter.h ual_defs.h ual_lowlevel.h ual_backend.h ual_context.h ual_exception.h ual_const.h \
 		$(includedir)
-	$(CP) latex html $(docdir)/dev/lowlevel
+	#$(CP) latex html $(docdir)/dev/lowlevel
 	$(INSTALL_PROGRAM) $(SPECIFIC_TOOLS) $(bindir)/.
 
 uninstall: libimas.so_uninstall libimas.a_uninstall pkgconfig_uninstall sources_uninstall
