--- qt-everywhere-opensource-src-5.7.0/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration_x11.cpp.orig	2016-06-10 08:48:56.000000000 +0200
+++ qt-everywhere-opensource-src-5.7.0/qtbase/src/plugins/platforms/offscreen/qoffscreenintegration_x11.cpp	2016-08-01 11:33:54.000000000 +0200
@@ -49,6 +49,7 @@
 
 #include <qpa/qplatformsurface.h>
 #include <qsurface.h>
+#include <stdio.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -229,7 +230,9 @@
 
 QFunctionPointer QOffscreenX11GLXContext::getProcAddress(const char *procName)
 {
-    return (QFunctionPointer)glXGetProcAddressARB(reinterpret_cast<const GLubyte *>(procName));
+  //return (QFunctionPointer)glXGetProcAddressARB(reinterpret_cast<const GLubyte *>(procName));
+  fprintf(stderr, "** QOffscreenX11GLXContext::getProcAddress() called, but no support compiled\n");
+  return NULL;
 }
 
 QSurfaceFormat QOffscreenX11GLXContext::format() const
