See https://bugreports.qt.io/browse/QTBUG-45139
--- qt-everywhere-opensource-src-5.5.0/qtbase/src/3rdparty/forkfd/forkfd.c.orig	2015-09-19 13:12:31.000000000 +0200
+++ qt-everywhere-opensource-src-5.5.0/qtbase/src/3rdparty/forkfd/forkfd.c	2015-09-19 14:10:49.000000000 +0200
@@ -41,11 +41,11 @@
 #include <unistd.h>
 
 #ifdef __linux__
-#  if (defined(__GLIBC__) && (__GLIBC__ << 16) + __GLIBC_MINOR__ >= 0x207) || defined(__BIONIC__)
+#  if (defined(__GLIBC__) && (__GLIBC__ << 8) + __GLIBC_MINOR__ >= 0x207) || defined(__BIONIC__)
 #    include <sys/eventfd.h>
 #    define HAVE_EVENTFD  1
 #  endif
-#  if (defined(__GLIBC__) && (__GLIBC__ << 16) + __GLIBC_MINOR__ >= 0x209) || defined(__BIONIC__)
+#  if (defined(__GLIBC__) && (__GLIBC__ << 8) + __GLIBC_MINOR__ >= 0x209) || defined(__BIONIC__)
 #    define HAVE_PIPE2    1
 #  endif
 #endif
