--- qt-everywhere-opensource-src-5.5.0/qtbase/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp.orig	2015-06-29 22:04:38.000000000 +0200
+++ qt-everywhere-opensource-src-5.5.0/qtbase/src/platformsupport/fontdatabases/fontconfig/qfontconfigdatabase.cpp	2015-09-19 18:33:49.000000000 +0200
@@ -94,8 +94,8 @@
         return mapToQtWeightForRange(fcweight, FC_WEIGHT_BOLD, FC_WEIGHT_ULTRABOLD, QFont::Bold, QFont::ExtraBold);
     if (fcweight <= FC_WEIGHT_BLACK)
         return mapToQtWeightForRange(fcweight, FC_WEIGHT_ULTRABOLD, FC_WEIGHT_BLACK, QFont::ExtraBold, QFont::Black);
-    if (fcweight <= FC_WEIGHT_ULTRABLACK)
-        return mapToQtWeightForRange(fcweight, FC_WEIGHT_BLACK, FC_WEIGHT_ULTRABLACK, QFont::Black, maxWeight);
+//    if (fcweight <= FC_WEIGHT_ULTRABLACK)
+//        return mapToQtWeightForRange(fcweight, FC_WEIGHT_BLACK, FC_WEIGHT_ULTRABLACK, QFont::Black, maxWeight);
     return maxWeight;
 }
 
