diff --git a/config/config.UL.gfortran b/config/config.UL.gfortran
new file mode 100644
index 00000000..a6ca420b
--- /dev/null
+++ b/config/config.UL.gfortran
@@ -0,0 +1,49 @@
+# Test whether necessary variables are set
+GLI_HOME ?= ${EBROOTGLI}
+LD_GR    ?= -L${EBROOTGR}/lib -lgr
+LD_GKS   ?= -L${GLI_HOME} -lgks -lXmu -lX11 -lXt -ldl
+
+# DEFINES  += -DRECL_UNITS_PER_DOUBLE=8 -DNO_SAVE
+ifndef SOLPS_CPP
+DEFINES  += -DGFORTRAN
+endif
+
+CPP       = -cpp -traditional
+ifdef SOLPS_MPI
+FC        = mpif90
+else
+FC        = gfortran
+endif
+CC        = /usr/bin/gcc
+
+GFORTRAN_MAJOR_VERSION = $(shell ${FC} --version | head -1 | awk '{printf "%s ",$$NF,"\n"}' | cut -d '.' -f 1 )
+GFORTRAN_MINOR_VERSION = $(shell ${FC} --version | head -1 | awk '{printf "%s ",$$NF,"\n"}' | cut -d '.' -f 2 )
+DEFINES  += -DCOMPILER_MAJOR_VERSION=${GFORTRAN_MAJOR_VERSION} -DCOMPILER_MINOR_VERSION=${GFORTRAN_MINOR_VERSION}
+
+FCOPTS    = -fPIC -fno-second-underscore
+ifdef SOLPS_DEBUG
+FCOPTS   += -g -fcheck=all -fbacktrace -frecursive -ffpe-trap=denormal -ffpe-trap=invalid -ffpe-trap=zero -ftrapv -fsignaling-nans
+CFLAGS    = -g
+else
+FCOPTS   += -O2
+CFLAGS    = -O2
+endif
+
+# NAG     = ${LD_NAG}
+GKS       = ${LD_GR} ${LD_GKS}
+
+LDLIBS    = -L/usr/local/lib -L/usr/lib/x86_64-linux-gnu
+LDEXTRA   =
+
+# new
+
+INCMOD    = -B
+MOD       = mod
+OBJDEST   = -J${OBJDIR} -o $@
+
+COMPILE   = ${FC} -c ${FCOPTS} ${INCMOD}${OBJDIR}
+
+#!!!Local Variables:
+#!!! mode: Makefile
+#!!! End:
+
